### PR TITLE
Fix timelimit not correct for Black Mesa

### PR DESCRIPTION
**Pull request by PlayBoy31**
Black Mesa is particular and use timelimit in seconds instead of minutes

### CODE REVIEW

File: core/TimerSys.cpp [hunk 1]
```diff
@@ -107,7 +111,7 @@ class DefaultMapTimer :
 
 		extra_time /= 60;
 
-		mp_timelimit->SetValue(mp_timelimit->GetInt() + extra_time);
+		mp_timelimit->SetValue(GetMapTimeLimit() + extra_time);
```
Comment (peace-maker) [inline core/TimerSys.cpp:114]: So is it in seconds or minutes now? You're setting the convar to minutes here. Probably need `* 60`?

Comment (PlayBoy31) [reply core/TimerSys.cpp]: Hummm, in fact we must stay in seconds here and convert only extratime if mp_timelimit is in minutes

**Comment by peace-maker** [comment]
`#if` preprocessor conditions usually aren't indented in this codebase. Looks good otherwise.

File: core/TimerSys.cpp [hunk 2]
```diff
@@ -81,7 +81,11 @@ class DefaultMapTimer :
 
 	int GetMapTimeLimit()
 	{
+		#if SOURCE_ENGINE == SE_BMS
```
Comment (dvander) [inline core/TimerSys.cpp:84]: Instead, can we have a single declaration at the top:

```
    #if SOURCE_ENGINE == SE_BMS
    static constexpr int kMapTimeScaleFactor = 60;
    #else
    static constexpr int kMapTimeScaleFactor = 1;
    #endif  
```

Then this becomes:

    return mp_timelimit->GetInt() / kMapTimeScaleFactor;

File: core/TimerSys.cpp [hunk 3]
```diff
@@ -105,7 +109,9 @@ class DefaultMapTimer :
 			return;
 		}
 
+		#if SOURCE_ENGINE != SE_BMS
 		extra_time /= 60;
+		#endif
```
Comment (dvander) [inline core/TimerSys.cpp:114]: And this becomes:

    extra_time /= (60 / kMapTimeScaleFactor);

Comment (PlayBoy31) [reply core/TimerSys.cpp:114]: Nice, its more generic, like it. I set a new proposal

File: core/TimerSys.cpp (hunk 2 above)
Comment (PlayBoy31) [reply core/TimerSys.cpp]: approved

File: core/TimerSys.cpp [hunk 4]
```diff
@@ -69,6 +69,13 @@ class DefaultMapTimer :
 	public IConVarChangeListener
 {
 public:
+
+	#if SOURCE_ENGINE == SE_BMS
```
Comment (dvander) [inline core/TimerSys.cpp:73]: Thanks, this looks much cleaner! As Peace-Maker noted, #if should be in the first column, preprocessor directives technically are not supposed to be indented. Though, this works if you'd like:

    #     if SOURCE_ENGINE == SE_BMS

Comment (PlayBoy31) [reply core/TimerSys.cpp]: Ho sorry forget that ! no problem i correct

**Comment by dvander** [comment]
Looks great, thanks!

**Pull request merged**